sys/net/network_layer/ng_sixlowpan/frag/ng_sixlowpan_frag.c

```diff
@@ -80,17 +80,17 @@ static uint16_t _send_1st_fragment(ng_sixlowpan_netif_t *iface, ng_pktsnip_t *pk
                                    size_t payload_len, size_t datagram_size)
 {
     ng_pktsnip_t *frag;
-    uint16_t max_frag_size = _floor8(iface->max_frag_size -
-                                     (payload_len - datagram_size) -
-                                     sizeof(ng_sixlowpan_frag_t));
     uint16_t local_offset = 0;
+    int payload_diff = (datagram_size - payload_len);
+    /* virtually add compressed bytes and remove dispatches */
```

Comment:
Could you also document what these variables contain? `datagram_size` is the size of full IPv6 packet, I assume? `payload_len` is the size for the IPv6 payload?